modules/ppcp-sepa/src/SEPAGateway.php

```diff
@@ -0,0 +1,297 @@
+<?php
+/**
+ * The SEPA Payment Gateway
+ *
+ * @package WooCommerce\PayPalCommerce\SEPA
+ */
+
+declare(strict_types=1);
+
+namespace WooCommerce\PayPalCommerce\SEPA;
+
+use Exception;
+use Psr\Log\LoggerInterface;
+use WC_Order;
+use WC_Payment_Gateway;
+use WooCommerce\PayPalCommerce\ApiClient\Endpoint\Orders;
+use WooCommerce\PayPalCommerce\ApiClient\Factory\PurchaseUnitFactory;
+use WooCommerce\PayPalCommerce\Session\SessionHandler;
+use WooCommerce\PayPalCommerce\WcGateway\Exception\GatewayGenericException;
+use WooCommerce\PayPalCommerce\WcGateway\Gateway\PayPalGateway;
+use WooCommerce\PayPalCommerce\WcGateway\Gateway\ProcessPaymentTrait;
+use WooCommerce\PayPalCommerce\WcGateway\Gateway\TransactionUrlProvider;
+use WooCommerce\PayPalCommerce\WcGateway\Processor\OrderProcessor;
+use WooCommerce\PayPalCommerce\WcGateway\Processor\RefundProcessor;
+
+/**
+ * Class SEPAGateway
+ */
+class SEPAGateway extends WC_Payment_Gateway {
+
+	use ProcessPaymentTrait;
+
+	const ID = 'ppcp-sepa';
+
+	/**
+	 * PayPal Orders endpoint.
+	 *
+	 * @var Orders
+	 */
+	private $orders_endpoint;
```

Comment:
I would suggest to always add typehints since we are already on PHP 7.4 +
```suggestion
	private Orders $orders_endpoint;
```

Reply:
I agree with this suggestion; when possible, we should use the type hints.

Just want to point out that this is only possible for custom properties, and we cannot add a type hint for properties that are defined without a type in the base class